Stream input files instead of reading entire contents

chancez opened the pull request:
Also supports streaming input from stdin and removes the requirement for
of passing any files to faq, defaulting to stdin if no file args are
specified.

I removed the usage of linguist because it's not really helping and I felt my custom code to detect xml/yaml/json is sufficient, along with the existing filename extension based detection. We could continue to use linguist, but it would mean we need to read a lot more of the file's contents (maybe all of it?) and pass those contents to lingust, which makes things complicated since the bufio.Reader buffer is of limited length.

jzelinskie commented on pkg/faq/faq.go:
```diff
@@ -28,32 +24,25 @@ func ProcessEachFile(inputFormat string, files []File, program string, programAr
-					return err
-				}
-				itemNum++
+		itemNum := 1
```
itemNum isn't being used?

jzelinskie commented on pkg/faq/faq.go:
```diff
@@ -237,49 +221,53 @@ func detectFormat(file File) (formats.Encoding, error) {
 			}
 		}
+		// we found a non-whitespace character that isn't what we were
+		// looking for, so stop trying to detect the format.
```
Full sentences with caps and punctuation!

chancez replied:
Sure would be nice if the linter handled this!

jzelinskie commented on pkg/faq/file.go:
```diff
@@ -1,51 +1,43 @@
 	return info.path
 }
 
+// Close closes the file
```
.

jzelinskie commented on go.mod:
```diff
@@ -1,7 +1,7 @@
 
 require (
-	github.com/Azure/draft v0.16.0
+	github.com/Azure/draft v0.16.0 // indirect
```
It doesn't look like you got rid of this. I don't think we should get rid of it anyway because linguist is able to autodetect formats like TOML, which your naive matching doesn't work on.

chancez replied:
That basically means throwing away streaming for auto detection and adding a good bit of complexity, but I can try.

jzelinskie replied:
Is that true? We don't need to give linguist the whole file.

chancez commented:
Okay, I fixed the review comments and added back linguist detection. I rebased on top of #62 because I found a bug in decoding logic while handling this and wanted to fix that independently of this PR.
